### ISSUE TITLE
Duplicate definition of genSym on Windows with ghc 8.6.2

### ISSUE THREAD

**Issue by LeventErkok** [digital-asset/ghc-lib#64]
GHC 8.6.2, Windows

I get the following error:

```
GHC runtime linker: fatal error: I found a duplicate definition for symbol
   genSym
whilst processing object file
   C:\ProgramData\chocolatey\lib\ghc\tools\ghc-8.6.2\lib\ghc-8.6.2\libHSghc-8.6.2.a
The symbol was previously defined in
   C:\Users\appveyor\AppData\Roaming\cabal\x86_64-windows-ghc-8.6.2\ghc-lib-parser-0.20190423-Jx5GWxfkulIAO3nX3UdYko\HSghc-lib-parser-0.20190423-Jx5GWxfkulIAO3nX3UdYko.o
This could be caused by:
   * Loading two different object files which export the same symbol
   * Specifying the same object file twice on the GHCi command line
   * An incorrect `package.conf' entry, causing some object to be
     loaded twice.
<no location info>: error:
    ghc.exe: unable to load package `ghc-lib-parser-0.20190423'
```

Full log file here: https://ci.appveyor.com/project/LeventErkok/sbv/builds/24577463

I'm pretty sure the dependency on `ghc-lib` comes via `hlint` in this case. I'm using the latest release of both.

This started happening recently: I've a build from about 8 days ago that had no issues:https://ci.appveyor.com/project/LeventErkok/sbv/builds/24385938 (And I didn't make any changes in the meantime myself.)

**Comment by shayne-fletcher-da** [comment]
Thanks for the report @LeventErkok! We think we have a pretty good idea about how this came about and are readying a fix. In the meantime, I can prevail upon you for instructions using your project on how I can reproduce this for myself?

**Comment by shayne-fletcher-da** [comment]
You are absolutely right by the way : `ghc-lib-parser` is getting pulled in by `hlint`. That's a relatively recent change to `hlint` and fits with the timelines.

**Comment by shayne-fletcher-da** [comment]
We're having trouble reproducing the issue in sbv nonetheless are reasonably confident https://github.com/digital-asset/ghc-lib/pull/65 will address it.

**Comment by LeventErkok** [comment]
Thanks!

I think you can duplicate by cloning https://github.com/LeventErkok/sbv and then:

    cabal configure --enable-tests
    cabal build

That's essentially what the Appveyor build is doing. Does that not exhibit the problem for you?

**Comment by LeventErkok** [comment]
@cocreature The failure on Appveyor uses GHC 8.6.2; perhaps that matters?

**Comment by shayne-fletcher-da** [comment]
I have replicated the build failure.

**Comment by shayne-fletcher-da** [comment]
It would seem there were two problems : (1) `ghc-lib-parser` symbols colliding with `ghc` libs and (2) ghc-8.6.2 itself for the missing runtime symbol `refptr.n_capabilities`. 

(1) was addressed by https://github.com/digital-asset/ghc-lib/pull/65 released as `ghc-lib-parser-0.20190516` and (2) by using ghc-8.6.5.

Here's the stack configuration I used.
```
resolver : lts-13.21 #ghc-8.6.5

packages:
- .

extra-deps:
- ghc-lib-parser-0.20190516
- hlint-2.1.20
- haskell-src-exts-1.21.0
- haskell-src-exts-util-0.2.5
```
Hope this helps you out @LeventErkok!
